cloudpathlib/client.py

```diff
@@ -37,7 +37,7 @@ def __init__(self, local_cache_dir: Optional[Union[str, os.PathLike]] = None):
 
     def __del__(self) -> None:
         # make sure temp is cleaned up if we created it
-        if self._cache_tmp_dir is not None:
+        if hasattr(self, "_cache_tmp_dir") and self._cache_tmp_dir is not None:
```

Comment:
```suggestion
        if getattr(self, "_cache_tmp_dir", None) is not None:
```

Should also work and be a little more concise, I think?